apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -168,30 +187,40 @@ def get_query_server_config(name='beeswax', connector=None):
       activeEndpoint = cache.get("hiveserver2")
       if activeEndpoint is None:
         if HIVE_DISCOVERY_HS2.get():
-          zk = KazooClient(hosts=libzookeeper_conf.ENSEMBLE.get(), read_only=True)
-          zk.start()
-          znode = HIVE_DISCOVERY_HIVESERVER2_ZNODE.get()
-          LOG.info("Selecting up Hive server via the following node {0}".format(znode))
-          if zk.exists(znode):
-            hiveservers = zk.get_children(znode)
-            LOG.info("Available Hive Servers: {0}".format(hiveservers))
-            if not hiveservers:
-              raise PopupException(_('There is no running Hive server available'))
-            server_to_use = 0  # if CONF.HIVE_SPREAD.get() randint(0, len(hiveservers)-1) else 0
-            LOG.info("Selected Hive server {0}: {1}".format(server_to_use, hiveservers[server_to_use]))
+          hiveservers = get_zk_hs2()
+          LOG.debug("Available Hive Servers: {0}".format(hiveservers))
+          if not hiveservers:
+            raise PopupException(_('There is no running Hive server available'))
+          server_to_use = 0  # if CONF.HIVE_SPREAD.get() randint(0, len(hiveservers)-1) else 0
+          LOG.debug("Selected Hive server {0}: {1}".format(server_to_use, hiveservers[server_to_use]))
+          cache.set(
+            "hiveserver2",
+            json.dumps({
+              "host": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0],
+              "port": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[1]
+            })
+          )
+        else:
+          cache.set("hiveserver2", json.dumps({"host": HIVE_SERVER_HOST.get(), "port": HIVE_HTTP_THRIFT_PORT.get()}))
+      elif activeEndpoint is not None:
+      # Replace ActiveEndpoint if the current HS2 is down
+        hiveservers = get_zk_hs2()
+        if hiveservers is not None:
+          server_to_use = 0 # if CONF.HIVE_SPREAD.get() randint(0, len(hiveservers)-1) else 0
+          hs2_HostName = hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0]
```

Comment:
Python var case is with underscores, e.g. hs2_HostName --> hs2_host_name

Reply:
done